index.js

```diff
@@ -84,8 +84,8 @@ function parse(str, options) {
  */
 
 function serialize(name, val, options) {
-  var opt = options || {};
-  var enc = opt.encode || encode;
+  var opt = typeof options === 'object' ? options : {};
+  var enc = typeof opt.encode === 'function' ? opt.encode : encode;
```

Comment:
Right, but fall-through is not good behavior; it should error. It errored before, though with a not great error, but it should still error.